core/metacling/src/TCling.cxx

```diff
@@ -5884,8 +5884,12 @@ static void* LazyFunctionCreatorAutoloadForModule(const std::string& mangled_nam
    R__LOCKGUARD(gInterpreterMutex);
 
    static bool sFirstRun = true;
-   // We don't want to do directory_iterator several times because it contains HDD access
-   static std::set<std::string> sLibraries;
+   // sLibraies contains pair of sPaths[i] (eg. /home/foo/module) and library name (eg. libTMVA.so). The
+   // reason why we're separating sLibraries and sPaths is that we have a lot of
+   // dupulication in path, for example we have "/home/foo/module-release/lib/libFoo.so", "/home/../libBar.so", "/home/../lib.."
+   // and it's waste of memory to store the full path.
+   static std::vector< std::pair<uint32_t, std::string> > sLibraries;
+   static std::vector<std::string> sPaths;
```

Comment:
`sPaths` can be taking `const std::string&`. We can delete `ModulePaths` variable and make sure that the loop variable `Path` is `const std::string&`. That is likely to reduce some memory.

Reply:
ModulePaths contains all paths in PrebuiltModulePaths, but we don't want to remember them all because they might contain empty directory or ".". I think we need to have string copy here, it's intentional

Reply:
The `PrebuiltModulePaths` is controlled by us and almost certainly we will not have empty directories intentionally. Either way the `.` filtered/handled separately.

Reply:
PrebuiltModulePaths does not guarantee that the directory is not empty. I can make an empty directory and add its path to LD_LIBRARY_PATH.

I'm not sure what you're suggesting, are you suggesting that we should delete path from PrebuiltModulePath rathe than adding valid path to sPaths?